src/octopeer-github/main/Options/Options.ts

```diff
@@ -10,75 +10,48 @@
-            self.doNotWatchHoverEvents = object.doNotWatchHoverEvents;
-            self.doNotWatchCommentElements = object.doNotWatchCommentElements;
-            self.doNotWatchKeyboardShortcutEvents = object.doNotWatchKeyboardShortcutEvents;
-            self.notifyObservers();
-        });
+        let options: string[] = this.generateOptionList();
+        chrome.storage.sync.get(options, this.syncOptionMap);
+        chrome.storage.onChanged.addListener((obj, area) => {if (area === "sync") {this.syncOptionMap(obj); }});
     }
 
     /**
-     * Enables a listener that listens for changes in the sync storage area.
-     * This means that any items that was changed (newValue) is set if changed;
+     * Handles synchronization of the optionMap by retrieving values from the chrome storage on initialization and on update.
+     * @param changeObject set of chrome storage properties.s
      */
-    public update() {
-        const self = this;
-        chrome.storage.onChanged.addListener(function (changes, areaName) {
-            if (areaName === "sync") {
-                const changeObject = <any> changes;
-                self.logging = changeObject.loggingEnabled ? changeObject.loggingEnabled.newValue : self.logging;
-                self.tabs = changeObject.trackTabs ? changeObject.trackTabs.newValue : self.tabs;
-                self.comments = changeObject.trackComments ? changeObject.trackComments.newValue : self.comments;
-                self.peerComments = changeObject.trackPeerComments ? changeObject.trackPeerComments.newValue : self.peerComments;
-                self.focus = changeObject.trackFocus ? changeObject.trackFocus.newValue : self.focus;
-                self.username = changeObject.hashUsername ? changeObject.hashUsername.newValue : self.username;
-                self.repo = changeObject.hashRepo ? changeObject.hashRepo.newValue : self.repo;
-                self.file = changeObject.hashFile ? changeObject.hashFile.newValue : self.file;
-                self.doNotWatchOnScreenEvents = changeObject.doNotWatchOnScreenEvents ?
-                    changeObject.doNotWatchOnScreenEvents.newValue : self.doNotWatchOnScreenEvents;
-                self.doNotWatchHoverEvents = changeObject.doNotWatchHoverEvents ?
-                    changeObject.doNotWatchHoverEvents.newValue : self.doNotWatchHoverEvents;
-                self.doNotWatchCommentElements = changeObject.doNotWatchCommentElements ?
-                    changeObject.doNotWatchCommentElements.newValue : self.doNotWatchCommentElements;
-                self.doNotWatchKeyboardShortcutEvents = changeObject.doNotWatchKeyboardShortcutEvents ?
-                    changeObject.doNotWatchKeyboardShortcutEvents.newValue : self.doNotWatchKeyboardShortcutEvents;
-                self.notifyObservers();
+    public syncOptionMap (changeObject: any) {
+        // For every option in optionMap, assign only the new values from the storage to them.
```

Comment:
As you might know I am no fan of inline comments (especially in such small methods)